src/js/lib/monster.ui.js

```diff
@@ -741,67 +741,187 @@ define(function(require) {
 		},
 
 		dialog: function(content, options) {
-			var isPersistent = _.has(options, 'isPersistent')
-				? options.isPersistent
-				: false;
-			var dialog = $('<div />').append(content),
-				coreApp = monster.apps.core,
-				i18n = coreApp.i18n.active(),
-				dialogType = typeof options !== 'undefined' && typeof options.dialogType !== 'undefined' ? options.dialogType : 'classic',
-				closeBtnText = i18n.close || 'X';
+			// Get options
+			var dialogType = _.get(options, 'dialogType', 'classic');
+			var isPersistent = _.get(options, 'isPersistent', false);
+			var minHeight = _.get(options, 'minHeight', 400);
+			var minWidth = _.get(options, 'minWidth', 400);
+			var scrollableContainer = options.scrollableContainer;
+			// Other variables/functions for internal use
+			var $dialogBody = $('<div />').append(content);
+			var $window = $(window);
+			var $body = $('body');
+			var coreApp = monster.apps.core;
+			var i18n = coreApp.i18n.active();
+			var closeBtnText = i18n.close || 'X';
+			var getElementSize = function($element) {
+				return {
+					width: $element.width(),
+					height: $element.height()
+				};
+			};
+			var windowLastSize = getElementSize($window);
+			var setDialogSizes = function() {
+				var dialogMaxHeight = $window.height() - 48;	// 100% - 3rem
+				var dialogMaxWidth = $window.width() - 48;	// 100% - 3rem
+				var dialogHeight = $dialog.height();
+				var dialogWidth = $dialog.width();
+				var $parentContainer = $scrollableContainer;
+				var dialogHeightDiff;
+				var dialogWidthDiff;
+
+				// Calculate maxs
+				if (dialogMaxHeight < minHeight) {
```

Comment:
I was wondering about it. Should `100% - 3rem` be always the upper boundary, even if the window is too small?

Reply:
We don't support mobile devices and most apps are somewhat tablet friendly so if a developer runs into that use case, it would be on them to declare a `scrollableContainer` that makes sense and plays nice with the dialog behavior.